packages/block-library/src/index.js

```diff
@@ -120,7 +121,10 @@ export const registerCoreBlocks = () => {
 		if ( ! block ) {
 			return;
 		}
-		const { name, settings } = block;
+		const { metadata, settings, name = metadata.name } = block;
```

Comment:
I suspect this may throw an error if a block exports neither `name` nor `metadata` (attempt to access property `name` of `undefined`).

Probably an issue regardless, but one which would be caught by `registerBlockType`, rather than being an error uncaught.

Reply:
Right, I can make it less magic to account for that.

Reply:
> Right, I can make it less magic to account for that.

At some point, I assume we'll want it more like how you had it originally where a block could be "manifest only" in its registration. For now, I think it's fine as long as every block we change to opt in to this workflow also exports its own name.

Reply:
In the current shape of RFC, I understand that you would always register blocks with JSON metadata which inside would reference all the files necessary to make block work. I started extracting all `icon` fields represented as SVG and `edit` implementations to their own files in #14743 to see how viable this approach could be. I plan to follow-up with all other fields described in RFC like `save` or `transform`.